dwpicker/selection.py

```diff
@@ -15,33 +13,26 @@ def select_targets(shapes, selection_mode='replace'):
     hovered = [s for s in shapes if s.hovered]
     targets = [t for s in hovered for t in s.targets() if cmds.objExists(t)]
 
-    if selection_mode in ('add', 'replace', 'invert'):
-        try:
-            return cmds.select(list(targets), add=selection_mode == 'add')
-        except ValueError:
-            raise NameclashError(targets)
+    clashing_nodes = [node for node in targets if len(cmds.ls(node)) > 1]
+    if clashing_nodes:
+        raise NameclashError(clashing_nodes)
+
+    current_selection = set(cmds.ls(selection=True, long=True))
+    targets_selection = set(cmds.ls(targets, long=True))
+    if selection_mode == 'add':
+        new_selection = current_selection | targets_selection
+    elif selection_mode in ('replace', 'invert'):
+        new_selection = targets_selection
     elif selection_mode == 'remove':
-        selection = [n for n in cmds.ls(sl=True) if n not in targets]
-        try:
-            return cmds.select(selection)
-        except ValueError:
-            raise NameclashError(targets)
-
-    # Invert selection
```

Comment:
I've removed the code bellow as it's dead code. This can't be called as `selection_mode` can only be `add`, `replace`, `invert` or `remove`.